pyiceberg/catalog/__init__.py

```diff
@@ -74,6 +75,8 @@
 LOCATION = "location"
 EXTERNAL_TABLE = "EXTERNAL_TABLE"
 
+TABLE_METADATA_FILE_NAME_REGEX = re.compile(r"""(\d+)-.*\.json""", re.X)
```

Comment:
To better avoid potential false positives, should we change this to:
```suggestion
TABLE_METADATA_FILE_NAME_REGEX = re.compile(r"""(\d+)-(.*)\.metadata\.json""", re.X)
```
Where the 2nd capturing group will also allow us to assert UUID validity later. I'm assuming we don't want to match non-UUID strings or arbitrary JSON files, but let me know if you think this is too restrictive.

Also, since we've specified the `re.X` flag anyway, we can make use of it by adding some comments here like:
```suggestion
TABLE_METADATA_FILE_NAME_REGEX = re.compile(
    r"""
    (\d+)             # version number
    -                 # separator
    (.*)              # UUID
    \.metadata\.json  # file extension
    """, 
    re.X
)
```

Reply:
Thanks for the suggestion. I've updated the regex to capture the UUID. Although in java we do not check the UUID validity: https://github.com/apache/iceberg/blob/81bf8d30766b1b129b87abde15239645cb127046/core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java#L382-L404
I think it make sense to add the check here